homeassistant/components/system_log/__init__.py

```diff
@@ -96,25 +97,67 @@ def _exception_as_string(exc_info):
     return buf.getvalue()
 
 
+class LogEntry:
+    """Store HA log entries."""
+
+    def __init__(self, record, stack, source):
+        """Initialize a log entry."""
+        self.timestamp = record.created
+        self.level = record.levelname
+        self.message = record.getMessage()
+        self.exception = _exception_as_string(record.exc_info)
+        self.source = source
+        self.count = 1
+
+    def hash(self):
+        """Calculate a key for DedupStore."""
+        key = hashlib.sha1()
+        key.update(self.level.encode('utf-8'))
+        key.update(self.message.encode('utf-8'))
```

Comment:
Not sure if we should include message, as they are sometimes customized to the message

Reply:
Not sure I follow... I though the all point was to dedup the message :/

Reply:
haha yes, but we should group them by stack trace. Because some exceptions will be: `Invalid value: yo`. Then if another invalid data comes in, the error might be `Invalid value: bleh`. Now those 2 are at the same place in the code, so it might make more sense to group them too. If we group on message, that won't happen.

Reply:
I was under the impression that the stack trace happened only once (it created the exception and that was it for that component/platform)

I do see you point, I have some ideas on how to work that out, but it will necessarily be probabilistic.

Reply:
Can't we just base the key on https://docs.python.org/3.5/library/traceback.html#traceback.extract_tb converted to a tuple ?

Reply:
I think we should stay away from that dedup based on the line the log was issued, while testing I realise that that approach deduplicates warnings related to loading multiple custom_components. 

I think we better not deduplicate that kind of information!

Reply:
We could whitelist certain warnings? 

Whitelisting on message means we group two JSON errors in different places?

Reply:
I think this approach is OK and works both cases.

I simplified the OrderedDict key as I didn't find any situation where the same message would appear with different log levels or source. What do you think ?